onnx/reference/ops/op_string_split.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) ONNX Project Contributors
+
+# SPDX-License-Identifier: Apache-2.0
+# pylint: disable=R0912,R0913,W0221
+
+from typing import Union
+
+import numpy as np
+
+from onnx.reference.op_run import OpRun
+
+_acceptable_str_dtypes = ("U", "O")
+
+
+def pad_empty_string(
+    split_lists: Union[list, np.ndarray], padding_requirement: Union[list, int]
+):
+    # pylint: disable=unidiomatic-typecheck`
+    if type(split_lists) is list:
+        return split_lists + ["" for _ in range(padding_requirement)]
+    elif type(split_lists) is np.ndarray:
+        return list(map(pad_empty_string, split_lists, padding_requirement))
+    else:
+        raise TypeError("Invalid array type")
+
+
+def split_with_padding(x, separator=None, maxsplit=None):
+    split_lists = np.char.split(x.astype(np.str_), separator, maxsplit)
```

Comment:
Does this require any special handling for the case with inputs of shape like `(2, 0)` ? To ensure that it returns a correct shape as per the ONNX spec, not as per np.char.split? Is there a test-case that covers it?

Reply:
There is a test case covering this now